### DISCUSSION TITLE
Cryostat and custom MXBeans

### DISCUSSION THREAD

**Discussion by grzesuav** [cryostatio/cryostat#621] [Q&A] [answered]
Hi, I can create custom MXBeans and view them in JMC - https://missioncontrol.mcnz.com/2020/12/10/MBeans_Management_Console_Mission_Control.html to switch some options etc.

Can cryostat support something similar ? Or is it already supported ?

**Comment by andrewazores** [comment, marked as answer]
We don't do anything like that at the moment. Cryostat only supports a small handful of MXBeans, just for those MBean Metrics dashboard cards essentially - no generic support for viewing or interacting with them.

It might be something we get around to implementing in the far future, but it wouldn't be a priority as Cryostat is focused more on JFR in particular and then more widely on broader monitoring/serviceability.

You might check out https://hawt.io/ for something like a "Cryostat for generic MBeans".

**Reply by grzesuav** [reply]
got it, thanks for the pointer
